Add timeout throttling example in setTimeout doc

jchimene opened the pull request:
Implement [timeout throttling example](https://github.com/mdn/sprints/issues/3692)

jchimene committed:
Update index.html

jchimene committed:
Capitalization

jchimene committed:
change capitalization in macro

jchimene committed:
Try heading co-ordination

jchimene committed:
Restore setTimeout() link

jchimene committed:
Update index.html

jpmedley commented on files/en-us/web/api/windoworworkerglobalscope/settimeout/index.html:
```diff
@@ -291,20 +291,90 @@ <h3 id="Passing_string_literals">Passing string literals</h3>
 <p>There are a number of reasons why a timeout may take longer to fire than anticipated. This section describes the most common reasons.</p>
 
-<h4 id="Timeouts_throttled_to_≥_4ms">Timeouts throttled to ≥ 4ms</h4>
+<h2 id="Timeouts">Timeouts throttled</h2>
```
This can't be an h2. Those need to be standard. Can you move this under "Example"?

jchimene replied:
I couldn't get the rest of the code example to work at h4. AFAICT, you have to descend one level after the <H2> to get HTML/JavaScript references to work in the `EmbedLiveSample` macro. The exit header is `Notes` at H2.
As a basis for this, I used the [Example at line 78](https://github.com/mdn/content/blob/main/files/en-us/web/api/windoworworkerglobalscope/settimeout/index.html) as a reference.
So, when you suggest moving to Example, does that mean the same line 78? and just refer to it in this section?

jchimene replied:
Closer to standard. Please inspect

jpmedley replied:
Down the road, we'll be implementing verification of page structure meaning this will eventually need to be addressed.

@chrisdavidmills I'm not sure which of your crew to include for help. Can you please pass this on?

sideshowbarker replied:
I’m going ahead and resolving this for the purposes of this PR. I guess for the general question, a new issue should be raised. Or if this PR as merged still exhibits the problem, then we can do a follow-up PR

jchimene replied:
> I’m going ahead and resolving this for the purposes of this PR. I guess for the general question, a new issue should be raised. Or if this PR as merged still exhibits the problem, then we can do a follow-up PR

I'm quite sure I addressed @jpmedley comment re: structure. After this comment, I moved & merged all examples as requested.

jchimene committed:
Move example to Examples section

Fixup the heading levels

jchimene committed:
Add Example link

jchimene committed:
Update index.html

More link goodness;
Remove explicit http: protocol references. This may not be SOP. Please confirm.

jchimene committed:
More work on the example

jchimene committed:
Update index.html

sideshowbarker commented:
Rebased. But there were a lot of conflicts, and I’m not sure I got it all right — so please check it very carefully.

jchimene commented:
> Rebased. But there were a lot of conflicts, and I’m not sure I got it all right — so please check it very carefully.

Yeah, thanks! I think this is the correct direction; the solution converged quite quickly. It's interesting to hone such a script to this environment.

jchimene committed:
Fixup EmbedLiveSample references

jchimene commented:
Hang on - I forgot the broken links.

jchimene committed:
fix: auto-edit broken links

jchimene committed:
Remove obsolete links
